CONTRIBUTING.md

```diff
@@ -1,59 +1,109 @@
+first tag. Note that (in most cases) this tag should be removed and Jenkins needs to be run before merging, to ensure
+the change does not break anything.
+
+The PR description describes (and in case of multiple commits, summarizes) the change in more detail.
+Again, try to describe the **why** (and in this case, to a lesser extent the **what**), rather than the **how**.
+
+If your PR is related to an open [issue](https://github.com/root-project/root/issues), make sure to link it.
+This will be done automatically if you add
+[closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
+to the PR description.
 
 Once a PR is created, a member of the ROOT team will review it as quickly as possible.  If you are familiar with the
 ROOT community, it may be beneficial to add a suggested reviewer to the PR in order to get quicker attention.
 Please ping people :wave: should you not get timely feedback, for instance with `@root-project/core ping!`
 
-Tests
------
+## Tests
 
 As you contribute code, this code will likely fix an issue or add a feature.
-Whatever it is: this requires you to add a new test, or to extend an existing test.
-We have concise unittests in the `test/` subdirectory of each part of ROOT;
-see for instance [`tree/dataframe/test`](https://github.com/root-project/root/tree/master/tree/dataframe/test).
-These tests are generally based on [Google Test](https://github.com/google/googletest) and easily extended.
-
-For more involved tests, such as tests requiring custom dictionaries or data
-files, we have [roottest](https://github.com/root-project/roottest.git).
-Suppose for your PR you create a branch on `root.git`.
-Our CI infrastructure automatically picks up a branch with the same name in your fork of `roottest.git`
-and use that for testing your PR.
-
-
-Continuous Integration
-----------------------
+Whatever it is: this requires you to add a new test, or to extend an existing test. Depending on the size and complexity
+of this test, it exists either in the `test/` subdirectory of each part of ROOT (see for instance
+[`tree/dataframe/test`](https://github.com/root-project/root/tree/master/tree/dataframe/test)), or in
+[roottest](https://github.com/root-project/roottest.git). Tests in `test/` subdirectories are unittests, mostly based on
+[Google Test](https://github.com/google/googletest) and easily extended. Tests in
+[roottest](https://github.com/root-project/roottest.git) are more involved (e.g., tests requiring custom dictionaries or
+data files). When you create a branch in the main ROOT repository (i.e., this repository) and add a test to `roottest`,
+make sure to do this under the same branch name (and open a PR for it). Our CI infrastructure automatically picks up the
+changes defined in the `roottest` PR based on this branchname, and uses that for testing your PR here.
```

Comment:
I like this addition about the tests! 

Maybe it's a corner case*  - but I stumbled upon this recently - that if you only add new tests in the roottest directory, in order to test if they don't break the new CI, you need to open a PR in the root repo (with some pseudo change, like an extra space in some description), using the same branch name as the one in the roottest, so that both CI's are tested. 

*and maybe more internal for ourselves in the team until roottest also tests for the new CI, if that's in the plans?

Reply:
> *and maybe more internal for ourselves in the team until roottest also tests for the new CI, if that's in the plans?

A PR adding proper roottest sync to the new CI is almost done/merged.

Reply:
While true, I believe that's unrelated to what Marta is mentioning here. Even with the sync of roottest when pushing PRs to root for the new CI, we still need to enable the new CI for those PRs that are pushed to roottest only, without a root sibling

Reply:
Indeed.  Once the source repository CI is fully function we should implement it for `roottest` (and it should not be too hard because it is probably similar to the source repository's)